packages/eslint-plugin-office-addins/src/rules/load-object-before-read.ts

```diff
@@ -0,0 +1,146 @@
+import { TSESTree } from "@typescript-eslint/typescript-estree";
+import {
+  Reference,
+  Scope,
+  Variable,
+} from "@typescript-eslint/experimental-utils/dist/ts-eslint-scope";
+import { isConstructorDeclaration } from "typescript";
+
+export = {
+  name: "load-object-before-read",
+  meta: {
+    type: <"problem" | "suggestion" | "layout">"problem",
+    messages: {
+      loadBeforeRead: "An explicit load call on '{{name}}' for '{{loadValue}}' needs to be made before reading a proxy object",
+    },
+    docs: {
+      description: 
+        "Before you can read the properties of a proxy object, you must explicitly load the properties",
+      category: <
+        "Best Practices" | "Stylistic Issues" | "Variables" | "Possible Errors"
+      >"Possible Errors",
+      recommended: <false | "error" | "warn">false,
+      url: 
+        "https://docs.microsoft.com/en-us/office/dev/add-ins/develop/application-specific-api-model?view=powerpoint-js-1.1#load",
+    },
+    schema: [],
+  },
+  create: function (context: any) {
+    function getPropertyThatHadToBeLoaded(node: TSESTree.Node): string | undefined {
+      if(node.parent?.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && node.parent?.property.type === TSESTree.AST_NODE_TYPES.Identifier) {
+        return node.parent.property.name;
+      }
```

Comment:
You're really doing all the testing on the parent node so you should pass that in instaed.

Reply:
Yes, I'm doing all the computations on the parent, which I think is also done on some other functions.
I could pass the parent beforehand, but then I will have to update the main code to see that and also check if the parent is not null when calling the function.

Do you thing it makes sense doing those steps everytime the functions is called? 

In general, I'm making the function receive a node reference everytime and this is pretty consistent accross all the functions in this file and I think this make it easier when calling the functions.

Reply:
You don't have to check the parent is null in the calling code . . . you can do it as part of the functions still.  See isNullObjectNode in the test-for-null-using-isNullObject rule.  

You are receiving a node either way . . . it's just a matter of which node you are looking at.  You can also type the input parameter to something more specific the just "node" and then you don't have to include that in the if statement.